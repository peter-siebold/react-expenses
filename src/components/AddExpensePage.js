import React from "react";
const AddExpensePage = () => (
    <div>Add new expenses</div>
)

export default AddExpensePage;