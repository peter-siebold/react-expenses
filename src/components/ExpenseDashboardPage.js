import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";
export const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpensesSummary />
        <ExpenseList />
    </div>
)

export default ExpenseDashboardPage;