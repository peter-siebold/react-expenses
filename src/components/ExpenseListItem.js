import React from "react";
import {connect} from "react-redux";
import { removeExpense } from "../actions/expenses";

const ExpenseListItem = ({description, amount, createdAt, id, dispatch}) => (
    <div >
        <h4>{description}</h4>
        <p>Amount: {amount} -  created at: {createdAt}</p>
        <button onClick={ () => {
            console.log(id)
            dispatch(removeExpense({id}))
            } }>Remove</button>
    </div>
);

export default connect()(ExpenseListItem);