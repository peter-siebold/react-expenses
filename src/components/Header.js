import React from "react";
import {NavLink} from "react-router-dom";

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink activeClassName="isActive" to="/" exact={true}>Dashboard</NavLink>
        <NavLink activeClassName="isActive" to="/create">Create Expense</NavLink>
    </header>
)

export default Header;