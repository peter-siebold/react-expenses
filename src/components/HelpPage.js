import React from "react";

const HelpPage = () => (
    <div>Help me please</div>
)

export default HelpPage;