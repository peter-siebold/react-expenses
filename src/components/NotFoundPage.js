import React from "react";
import {Link} from "react-router-dom";
export const NotFoundPage = () => (
    <div>
        404 - Not found <Link to="/">Back to Home</Link>
    </div>
)

export default NotFoundPage;