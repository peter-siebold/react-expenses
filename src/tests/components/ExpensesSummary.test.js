import React from "react";
import {ExpensesSummary} from "../../components/ExpensesSummary";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";



test("should render ExpensesSummary with 1 expense", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
})
test("should render ExpensesSummary with multiple expense", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={120} expensesTotal={235456456} />);
    expect(wrapper).toMatchSnapshot();
})
