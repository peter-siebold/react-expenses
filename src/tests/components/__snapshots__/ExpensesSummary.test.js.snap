// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling 
  $2.35
</div>
`;

exports[`should render ExpensesSummary with multiple expense 1`] = `
<div>
  Viewing 
  120
   
  expenses
   totalling 
  $2,354,564.56
</div>
`;
